src/en/koharu/src/eu/kanade/tachiyomi/extension/en/koharu/Koharu.kt

```diff
@@ -0,0 +1,279 @@
+            }
+            cosplayers.emptyToNull()?.joinToString()?.let {
+                append("Cosplayers: ", it, "\n")
+            }
+            parodies.emptyToNull()?.joinToString()?.let {
+                append("Parodies: ", it, "\n")
+            }
+            characters.emptyToNull()?.joinToString()?.let {
+                append("Characters: ", it, "\n")
+            }
+            append("Pages: ", thumbnails.entries.size, "\n\n")
+
+            try {
+                append("Added: ", dateReformat.format(((updated_at ?: created_at))), "\n")
+            } catch (_: Exception) {}
+        }
+        status = SManga.COMPLETED
+        update_strategy = UpdateStrategy.ONLY_FETCH_ONCE
+        initialized = true
+    }
+
+    private fun <T> Collection<T>.emptyToNull(): Collection<T>? {
+        return this.ifEmpty { null }
+    }
+
+    override fun getMangaUrl(manga: SManga) = "$baseUrl/g/${manga.url}"
+
+    // Chapter
+
+    override fun chapterListRequest(manga: SManga): Request {
+        return GET("$apiBooksUrl/detail/${manga.url}", headers)
+    }
+
+    override fun chapterListParse(response: Response): List<SChapter> {
+        val manga = response.parseAs<MangaEntry>()
+        return listOf(
+            SChapter.create().apply {
+                name = "Chapter"
+                url = "${manga.id}/${manga.public_key}"
+                date_upload = (manga.updated_at ?: manga.created_at)
+            },
+        )
+    }
+
+    override fun getChapterUrl(chapter: SChapter) = "$baseUrl/g/${chapter.url}"
+
+    // Page List
+
+    override fun pageListRequest(chapter: SChapter): Request {
+        return GET("$apiBooksUrl/detail/${chapter.url}", headers)
+    }
+
+    override fun pageListParse(response: Response): List<Page> {
+        val mangaEntry = response.parseAs<MangaEntry>()
+        val imagesInfo = getImagesByMangaEntry(mangaEntry)
+
+        return imagesInfo.entries.mapIndexed { index, image ->
+            Page(index, imageUrl = "${imagesInfo.base}/${image.path}")
+        }
+    }
+
+    override fun imageUrlParse(response: Response) = throw UnsupportedOperationException()
+
+    // Settings
+
+    override fun setupPreferenceScreen(screen: PreferenceScreen) {
+        ListPreference(screen.context).apply {
+            key = PREF_IMAGERES
+            title = "Image Resolution"
+            entries = arrayOf("780x", "980x", "1280x", "1600x", "Original")
+            entryValues = arrayOf("780", "980", "1280", "1600", "0")
+            summary = "%s"
+
+            setOnPreferenceChangeListener { _, newValue ->
+                quality = newValue.toString()
+                true
+            }
+        }.also(screen::addPreference)
+    }
+
+    @Suppress("UNCHECKED_CAST")
+    private fun readInstanceProperty(instance: Data, propertyName: String): Any? {
+        val property = instance::class.members.first { it.name == propertyName } as KProperty1<Any, *>
+        return property.get(instance)
+    }

```

Comment:
![thisisfine](https://cdn.discordapp.com/emojis/927962828294471681.webp?size=48&quality=lossless)

you can use java reflection which avoids adding another library?